providers/pyproject.toml

```diff
@@ -0,0 +1,98 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[project]
+name = "local-providers"
+version = "0.1.0"
+description = "Placeholder package for local/from-sources providers."
+requires-python = ">=3.8, <3.13"
```

Comment:
This purposefully stays at 3.8 in this PR, as if I change it to 3.9 then there are a load of upgrade diffs applied and I don't want to mix up this already massive PR with more changes than are strictly required.

Once it lands I will change this and apply all the upgrades it results in.